storage/innobase/log/log0log.cc

```diff
@@ -578,32 +580,29 @@ void log_t::resize_abort() noexcept
 
 /** Write an aligned buffer to ib_logfile0.
 @param buf    buffer to be written
-@param len    length of data to be written
+@param length length of data to be written
 @param offset log file offset */
-static void log_write_buf(const byte *buf, size_t len, lsn_t offset)
+static void log_write_buf(const byte *buf, size_t length, lsn_t offset)
 {
   ut_ad(write_lock.is_owner());
   ut_ad(!recv_no_log_write);
   ut_d(const size_t block_size_1= log_sys.get_block_size() - 1);
   ut_ad(!(offset & block_size_1));
-  ut_ad(!(len & block_size_1));
+  ut_ad(!(length & block_size_1));
   ut_ad(!(size_t(buf) & block_size_1));
-  ut_ad(len);
+  ut_ad(length);
+
+  const lsn_t maximum_write_length{log_sys.file_size - offset};
```

Comment:
Because `lsn_t` is not signed, a wrap-around in the subtraction is allowed.
```c++
ut_ad(maximum_write_length <= log_sys.file_size - log_sys.START_OFFSET);
```
would also work and be slightly stricter. This function is only being used for writing log records, and thus `offset >= log_sys.START_OFFSET` must hold.